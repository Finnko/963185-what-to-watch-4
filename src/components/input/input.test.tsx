import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Input from './input';

const props = {
  type: `radio`,
  groupName: `rating`,
  id: `1`,
  name: `email`,
  placeholder: `Email`,
  value: `a@a.ru`,
  label: `Email`,
  touched: true,
  disabled: false,
  shouldValidate: true,
  onInputChange: () => null,
};

describe(`Input component render correctly`, () => {
  it(`Should Input component render correctly when its valid`, () => {
    const tree = renderer
      .create(
          <Input valid {...props}/>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Input component render correctly when its invalid`, () => {
    const tree = renderer
      .create(
          <Input valid={false} {...props}/>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
