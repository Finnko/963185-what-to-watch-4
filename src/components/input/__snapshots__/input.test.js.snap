// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component render correctly Should Input component render correctly when its invalid 1`] = `
<div
  className="sign-in__field sign-in__field--error"
>
  <input
    className="sign-in__input"
    id="1"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="email"
    value="a@a.ru"
  />
  <label
    className="sign-in__label visually-hidden"
    htmlFor="1"
  >
    Email
  </label>
</div>
`;

exports[`Input component render correctly Should Input component render correctly when its valid 1`] = `
<div
  className="sign-in__field "
>
  <input
    className="sign-in__input"
    id="1"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="email"
    value="a@a.ru"
  />
  <label
    className="sign-in__label visually-hidden"
    htmlFor="1"
  >
    Email
  </label>
</div>
`;
